lib/ruby_smb/server/session.rb

```diff
@@ -62,6 +63,11 @@ def logoff!
       #   @return [Hash]
       attr_accessor :tree_connect_table
 
+      # Untyped hash for storing additional arbitrary metadata about the current session
+      # @!attribute [rw] metadaa
+      #   @return [Hash]
+      attr_accessor :metadata
```

Comment:
I went with `metadata` at the time of implementation, but on reflection I'm thinking this is closer to `state` or just `data` in general

Reply:
This isn't a blocker for me, it turns out framework uses `context` - but I'm happy to stick with `metadata` for now